pkg/proc/disasm_arm64_arch.go

```diff
@@ -0,0 +1,212 @@
+// +build linux,arm64
+
+package proc
+
+import (
+	"golang.org/x/arch/arm64/arm64asm"
+)
+
+var maxInstructionLength uint64 = 15
+
+type archInst struct {
+	Inst arm64asm.Inst
+	Len  int
+}
+
+func asmDecode(mem []byte, pc uint64) (*archInst, error) {
+	inst, err := arm64asm.Decode(mem)
+	if err != nil {
+		return nil, err
+	}
+
+	r := archInst{inst, 4}
+	return &r, nil
+}
+
+func (inst *archInst) Size() int {
+	return inst.Len
+}
+
+// Text will return the assembly instructions in human readable format according to
+// the flavour specified.
+func (inst *AsmInstruction) Text(flavour AssemblyFlavour, bi *BinaryInfo) string {
+	if inst.Inst == nil {
+		return "?"
+	}
+
+	var text string
+
+	switch flavour {
+	case GNUFlavour:
+		text = arm64asm.GNUSyntax(inst.Inst.Inst)
+	default:
+		text = arm64asm.GoSyntax(inst.Inst.Inst, inst.Loc.PC, bi.symLookup, nil)
+	}
+
+	return text
+}
+
+// IsCall returns true if the instruction is a CALL or LCALL instruction.
+func (inst *AsmInstruction) IsCall() bool {
+	if inst.Inst == nil {
+		return false
+	}
+	return inst.Inst.Inst.Op == arm64asm.BL || inst.Inst.Inst.Op == arm64asm.BLR
+}
+
+// IsRet returns true if the instruction is a RET or ERET instruction.
+func (inst *AsmInstruction) IsRet() bool {
+	if inst.Inst == nil {
+		return false
+	}
+	return inst.Inst.Inst.Op == arm64asm.RET || inst.Inst.Inst.Op == arm64asm.ERET
+}
+
+func resolveCallArg(inst *archInst, inst_addr uint64, currentGoroutine bool, regs Registers, mem MemoryReadWriter, bininfo *BinaryInfo) *Location {
+	if inst.Inst.Op != arm64asm.BL && inst.Inst.Op != arm64asm.BLR {
+		return nil
+	}
+
+	var pc uint64
+	var err error
+
+	switch arg := inst.Inst.Args[0].(type) {
+	case arm64asm.Imm:
+		pc = uint64(arg.Imm)
+	case arm64asm.Reg:
+		if !currentGoroutine || regs == nil {
+			return nil
+		}
+		pc, err = regs.Get(int(arg))
+		if err != nil {
+			return nil
+		}
+	case arm64asm.PCRel:
+		pc = uint64(inst_addr) + uint64(arg)
+	default:
+		return nil
+	}
+
+	file, line, fn := bininfo.PCToLine(pc)
+	if fn == nil {
+		return &Location{PC: pc}
+	}
+	return &Location{PC: pc, File: file, Line: line, Fn: fn}
+}
+
+type instrseq []arm64asm.Op
+
+// Possible stacksplit prologues are inserted by stacksplit in
+// $GOROOT/src/cmd/internal/obj/arm64/obj7.go.
+// The stacksplit prologue will always begin with loading curg in CX, this
+// instruction is added by load_g_cx in the same file and is either 1 or 2
+// MOVs.
+var prologues []instrseq
+
+func init() {
+	var tinyStacksplit = instrseq{arm64asm.MOV, arm64asm.CMP, arm64asm.B}
+	var smallStacksplit = instrseq{arm64asm.SUB, arm64asm.CMP, arm64asm.B}
+	var bigStacksplit = instrseq{arm64asm.CMP, arm64asm.B, arm64asm.ADD, arm64asm.SUB, arm64asm.MOV, arm64asm.CMP, arm64asm.B}
+	var unixGetG = instrseq{arm64asm.MOV}
+	//var windowsGetG = instrseq{arm64asm.MOV, arm64asm.MOV}
```

Comment:
Why comment this out?

Reply:
Because go arm64 do not support windows for now.(go1.12)

Reply:
Let's remove it then instead of just commenting it out until it's supported.

Reply:
modified.